src/include/miopen/type_name.hpp

```diff
@@ -34,11 +34,8 @@ namespace miopen {
 template <class MIOpen_Private_TypeName_>
 const std::string& get_type_name()
 {
-    // NOLINTNEXTLINE (cppcoreguidelines-avoid-non-const-global-variables)
-    static std::string name;
-
-    if(name.empty())
-    {
+    static const std::string ret = []() {
```

Comment:
Since this relies on `__PRETTY_FUNCTION__` to get the type name then this can't be put into a lambda. I think you will need to make a seperate `compute_type_name` function and then call it. Thats what we do [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/type_name.hpp#L34).

Reply:
Good catch, Paul! but let me do that separately. This is a bit out of scope of this PR and I would like to look how this stuff is used.

Reply:
Sure, but we have unit tests here that could fail:

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/test/type_name.cpp#L24

If the tests do pass then you can ignore my comment.

Reply:
Indeed, in spite of `__PRETTY_FUNCTION__` outputs full signature, including the caller's name (https://reviews.llvm.org/D33497) the test fails. Let me fix this.

Reply:
Resolved!